dask/array/tests/test_reductions.py

```diff
@@ -283,7 +283,9 @@ def test_nanarg_reductions(dfunc, func):
 @pytest.mark.parametrize("func", ["argmax", "nanargmax"])
 def test_arg_reductions_unknown_chunksize(func):
     x = da.arange(10, chunks=5)
-    x = x[x > 1]
+    y = x[x > 1]
+    y._chunks = ((np.nan,),)
+    x = y
```

Comment:
Similar comment here:

```suggestion
    x._chunks = tuple((np.nan,) * len(c) for c in x.chunks)
```